### PR TITLE
Feature/1387 tag component

### PR DESCRIPTION
**Pull request by anna-lach**

### CODE REVIEW

File: packages/components/tag/src/tag-list.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
+import { localized, msg } from '@lit/localize';
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { RovingTabindexController } from '@sl-design-system/shared';
+import { Tooltip } from '@sl-design-system/tooltip';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html, nothing } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './tag-list.scss.js';
+import { Tag, TagEmphasis, TagSize } from './tag.js';
```
Comment (jpzwarte) [inline packages/components/tag/src/tag-list.ts:8]: Shouldn't `TagEmphasis` and `TagSize` be `type` imports?

File: packages/components/tag/src/tag-list.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.setAttribute('role', 'list');
+
+    // We need to wait for the next frame so the element has time to render
+    requestAnimationFrame(() => {
+      this.#resizeObserver.observe(this);
+
+      this.tags?.forEach((tag: Tag) => (tag.size = this.size));
```
Comment (jpzwarte) [inline packages/components/tag/src/tag-list.ts:114]: The `Tag` explicit type is unnecessary here.

File: packages/components/tag/src/tag-list.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    this.#mutationObserver.disconnect();
+
+    super.disconnectedCallback();
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('size')) {
+      this.tags?.forEach((tag: Tag) => (tag.size = this.size));
```
Comment (jpzwarte) [inline packages/components/tag/src/tag-list.ts:133]: Same unnecessary `Tag` type

File: packages/components/tag/src/tag-list.ts [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    }
+
+    if (changes.has('emphasis')) {
+      this.tags?.forEach((tag: Tag) => (tag.emphasis = this.emphasis));
```
Comment (jpzwarte) [inline packages/components/tag/src/tag-list.ts:137]: Same unnecessary `Tag` type

File: packages/components/tag/src/tag-list.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      ${this.stacked && this.#hiddenLabel > 0
+        ? html` <div class="group" tabindex="-1">
```
Comment (jpzwarte) [inline packages/components/tag/src/tag-list.ts:144]: Please for a newline after the opening backtick

**Review by jpzwarte — APPROVED** [review]
I've only partially reviewed this (busy at the office atm :/ )

File: packages/components/tag/src/tag-list.scss [hunk 6]
```diff
@@ -0,0 +1,104 @@
+:host {
+  --_gap: var(--sl-space-tag-list-md);
+  --_group-font: var(--sl-text-tag-label-md);
+  --_group-margin: calc(var(--_group-space) * 4);
+  --_group-margin-more: calc(var(--_group-space) * 2);
+  --_group-space: var(--sl-space-tag-counter-gap);
+  --_group-width: var(--sl-size-tag-counter);
+
+  display: flex;
+  flex-flow: column;
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/tag/src/tag-list.scss:10]: good one, I didn't see that mess 😆

File: packages/components/tag/src/tag-list.scss [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+:host([size='lg']) {
+  --_stacked-height: var(--sl-size-tag-lg);
+}
+
+.group {
+  display: block;
+
+  sl-tag {
+    display: block;
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/tag/src/tag-list.scss:40]: by mistake :)

File: packages/components/tag/src/tag-list.ts [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+  interface GlobalEventHandlersEventMap {
+    'sl-tab-change': SlTabChangeEvent;
+  }
+
+  interface HTMLElementTagNameMap {
+    'sl-tag-list': TagList;
+  }
+}
+
+export type SlTabChangeEvent = CustomEvent<number>;
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/tag/src/tag-list.ts:20]: left by copying, I'll remove it :)

File: packages/components/tag/src/tag.stories.ts [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+      options: emphases
+    },
+    size: {
+      control: 'inline-radio',
+      options: sizes
+    }
+  },
+  parameters: {
+    // Notifies Chromatic to pause the animations at the first frame for this specific story.
+    chromatic: { pauseAnimationAtEnd: false, prefersReducedMotion: 'reduce' }
```
Comment (Diaan) [inline packages/components/tag/src/tag.stories.ts:36]: are there animations here?

File: packages/components/tag/register.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+import { Tag } from './src/tag.js';
```
Comment (Diaan) [inline packages/components/tag/register.ts:1]: sorting should be the other way around

File: packages/components/tag/src/tag.ts [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /** The emphasis of the tag; defaults to 'subtle'. */
+  @property({ reflect: true }) emphasis: TagEmphasis = 'subtle';
+
+  /** The label of the tag component. */
+  @property({ reflect: true }) label?: string;
+
+  /** The size of the tag. Defaults to `md`. */
+  @property({ reflect: true }) size?: TagSize = 'md';
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/tag/src/tag.ts:64]: So for example like we have in a `button-group` with size, yes?

https://github.com/sl-design-system/components/pull/1410/files

File: packages/components/tag/src/tag.stories.ts (hunk 9 above)
Comment (anna-lach) [reply packages/components/tag/src/tag.stories.ts:36]: no, just left by copying from other component

**Review by Diaan — APPROVED** [review]

**Pull request merged**